build.sbt

```
@@ -1,53 +1,94 @@
+
+val akkaActorVersion = "2.6.17"
+val akkaHttpVersion = "10.2.6"
+
+inThisBuild(
+  List(
+    versionPolicyIntention := Compatibility.BinaryAndSourceCompatible,
+    organization := "org.endpoints4s",
+    sonatypeProjectHosting := Some(
+      GitHubHosting("endpoints4s", "akka-http", "julien@richard-foy.fr")
+    ),
+    homepage := Some(sonatypeProjectHosting.value.get.scmInfo.browseUrl),
+    licenses := Seq(
+      "MIT License" -> url("http://opensource.org/licenses/mit-license.php")
+    ),
+    developers := List(
+      Developer(
+        "julienrf",
+        "Julien Richard-Foy",
+        "julien@richard-foy.fr",
+        url("http://julien.richard-foy.fr")
+      )
+    ),
+    scalaVersion := "2.13.8",
+    crossScalaVersions := Seq("2.13.8", "3.0.2", "2.12.13"),
+    versionPolicyIgnoredInternalDependencyVersions := Some("^\\d+\\.\\d+\\.\\d+\\+\\d+".r)
+  )
+)
 
 val `akka-http-client` =
   project
     .in(file("client"))
     .settings(
-      publishSettings,
-      `scala 2.12 to dotty`,
       name := "akka-http-client",
       publish / skip := scalaBinaryVersion.value.startsWith("3"),
       libraryDependencies ++= Seq(
-        ("com.typesafe.akka" %% "akka-stream" % akkaActorVersion % Provided).cross(CrossVersion.for3Use2_13),
+        algebra,
+        openapi,
+        algebraTestkit % Test,
+        algebraCirceTestkit % Test,
+        jsonSchemaGeneric % Test,
+        ("com.typesafe.akka" %% "akka-stream" % akkaActorVersion % Provided).cross(
+          CrossVersion.for3Use2_13
+        ),
         ("com.typesafe.akka" %% "akka-http" % akkaHttpVersion).cross(CrossVersion.for3Use2_13),
-        ("com.typesafe.akka" %% "akka-http-testkit" % akkaHttpVersion % Test).cross(CrossVersion.for3Use2_13),
-        ("com.typesafe.akka" %% "akka-stream-testkit" % akkaActorVersion % Test).cross(CrossVersion.for3Use2_13),
-        scalaTestDependency
+        ("com.typesafe.akka" %% "akka-http-testkit" % akkaHttpVersion % Test).cross(
+          CrossVersion.for3Use2_13
+        ),
+        ("com.typesafe.akka" %% "akka-stream-testkit" % akkaActorVersion % Test).cross(
+          CrossVersion.for3Use2_13
+        )
       ),
       excludeDependencies ++= {
         if (scalaBinaryVersion.value.startsWith("3")) {
           List(ExclusionRule("org.scala-lang.modules", "scala-collection-compat_2.13"))
         } else Nil
       }
     )
-    .dependsOn(`algebra-jvm`)
-    .dependsOn(`openapi-jvm`)
-    .dependsOn(`algebra-testkit-jvm` % Test)
-    .dependsOn(`algebra-circe-testkit-jvm` % Test)
-    .dependsOn(`json-schema-generic-jvm` % "test->test")
 
 val `akka-http-server` =
   project
     .in(file("server"))
     .settings(
-      publishSettings,
-      `scala 2.12 to dotty`,
       name := "akka-http-server",
       publish / skip := scalaBinaryVersion.value.startsWith("3"),
       libraryDependencies ++= Seq(
+        algebra,
+        openapi,
+        algebraTestkit % Test,
```

If that does not work with Scala 3, it should be possible to use the `CrossVersion.for3Use2_13` trick, no?

```suggestion
        (algebraTestkit % Test).cross(CrossVersion.for3Use2_13),
```

---

That does not work - the Scala 3 `compile` starts throwing various Scala 3 errors with `CrossVersion.for3Use2_13`.

---

Please check latest CI run for errors.